_posts/2018-01-07-building-serverless-radiator.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: post
+title: Building a serverless radiator for fun and benefit with Clojure, Python and AWS 
+author: hjhamala
+excerpt: Monitoring software development pipeline statuses, environment alarms and metrics are an essential part of a modern software development process in a cloud environment. In this post I show a simple way to monitor different accounts with a serverless radiator. 
+tags:
+- AWS
+- Clojure
+- Serverless 
+---
+
+***There exists many commercial products for monitoring development pipelines and deployments. What could be more fun than make your own?***
+
+**Modern** software development environment is most likely composed of multiple pipelines. There can be pipelines for build artifacts, infrastructure building etc. It is vital to notice fast when the builds fail. Monitoring only the pipelines is not necessary enough. Also deployments should be monitored for noticing potential problems in performance etc. 
+
+One project might have different accounts for development, testing/integration and production. This means that monitoring simply one account is not even closely enough. For example in my current project at Solita, we have 3 different accounts which have 7 pipelines, over 60 CloudWatch alarms and some exposed metrics from deployments.
+
+One good solution is to have a radiator which shows development status:
+
+![Alarm!](/img/building-serverless-radiator/complete-radiator.png)
+
+Because making your own is interesting and fun I chose to make my own radiator. For cost optimization I chose serverless approach with AWS [Lambdas](https://aws.amazon.com/lambda/). Lets start our journey from how to expose pipelines, alarms and metrics from AWS accounts.
+
+## Exposing environment statuses from AWS accounts
+
+AWS [CodePipeline](https://aws.amazon.com/codepipeline/) is an AWS software product for building, testing and installing software to the AWS environment. It integrates very nicely with AWS [CloudFormation](https://aws.amazon.com/cloudformation/) and is a pay only what you use product. There is no need to provision EC2 virtual machines for builds - AWS will provision them for you.
+
+One can get statuses from CodePipeline by using AWS CLI or AWS SDK which is available for many different languages. AWS [CloudWatch](https://aws.amazon.com/cloudwatch/) alarms and metrics are also accessible via CLI and SDK. 
+
+One way to extract data would be to make an AWS user to every account, generate a access key id and a secret key and poll end points directly from the radiator software. A variant of this would be to make a AWS role and configure cross account permissions to switch to that role. 
```

Comment:
"I chose *a* serverless approach…"

"Let's start…" (heittomerkki puuttuu)

"…our journey by seeing how to" voisi olla idiomaattisempi kuin "from".

"is a pay-only-what-you-use product"

"generate *an* access key ID"